galley/pkg/config/analysis/diag/messages_test.go

```diff
@@ -114,11 +112,74 @@ func TestMessages_SortedCopy(t *testing.T) {
 	g.Expect(newMsgs).To(Equal(expectedMsgs))
 }
 
-func testResource(name string) *resource.Instance {
-	return &resource.Instance{
-		Metadata: resource.Metadata{
-			FullName: resource.NewShortOrFullName("default", name),
-		},
-		Origin: testOrigin{name: name},
+func TestMessages_SetRefDoc(t *testing.T) {
+	g := NewWithT(t)
+
+	firstMsg := NewMessage(
+		NewMessageType(Error, "B1", "Template: %q"),
+		MockResource("B"),
+		"B",
+	)
+	secondMsg := NewMessage(
+		NewMessageType(Info, "C1", "Template: %q"),
+		MockResource("B"),
+		"B",
+	)
+
+	msgs := Messages{firstMsg, secondMsg}
+	msgs.SetDocRef("istioctl-awesome")
+
+	getDocURL := func(msg Message) string {
+		return msg.Unstructured(false)["documentation_url"].(string)
 	}
+
+	g.Expect(getDocURL(msgs[0])).To(Equal("https://istio.io/docs/reference/config/analysis/b1/?ref=istioctl-awesome"))
+	g.Expect(getDocURL(msgs[1])).To(Equal("https://istio.io/docs/reference/config/analysis/c1/?ref=istioctl-awesome"))
+}
+
+func TestMessages_Filter(t *testing.T) {
+	g := NewWithT(t)
+
+	firstMsg := NewMessage(
+		NewMessageType(Error, "B1", "Template: %q"),
+		MockResource("B"),
+		"B",
+	)
+	secondMsg := NewMessage(
+		NewMessageType(Info, "A1", "Template: %q"),
+		MockResource("B"),
+		"B",
+	)
+	thirdMsg := NewMessage(
+		NewMessageType(Warning, "C1", "Template: %q"),
+		MockResource("B"),
+		"B",
+	)
+
+	msgs := Messages{firstMsg, secondMsg, thirdMsg}
+	filteredMsgs := msgs.FilterOutLowerThan(Warning)
+	expectedMsgs := Messages{firstMsg, thirdMsg}
+
+	g.Expect(filteredMsgs).To(Equal(expectedMsgs))
+}
+
+func TestMessages_FilterOutAll(t *testing.T) {
+	g := NewWithT(t)
+
+	firstMsg := NewMessage(
+		NewMessageType(Info, "A1", "Template: %q"),
+		MockResource("B"),
+		"B",
+	)
+	secondMsg := NewMessage(
+		NewMessageType(Warning, "C1", "Template: %q"),
+		MockResource("B"),
+		"B",
+	)
+
+	msgs := Messages{firstMsg, secondMsg}
+	filteredMsgs := msgs.FilterOutLowerThan(Error)
+	expectedMsgs := Messages{}
+
+	g.Expect(filteredMsgs).To(Equal(expectedMsgs))
 }
```

Comment:
Not necessary in this PR if you don't want to, but I think this would be a good candidate for table driven tests. See https://github.com/golang/go/wiki/TableDrivenTests